appyx-navigation/common/src/commonMain/kotlin/com/bumble/appyx/navigation/composable/AppyxComponent.kt

```diff
@@ -132,65 +139,92 @@ class ChildrenTransitionScope<InteractionTarget : Any, NavState : Any>(
+                                        .pointerInput(appyxComponent) {
+                                            detectDragGesturesOrCancellation(
+                                                onDragStart = { position ->
+                                                    appyxComponent.onStartDrag(position)
+                                                },
+                                                onDrag = { change, dragAmount ->
+                                                    if (gestureValidator.isGestureValid(
+                                                            change.position,
+                                                            transformedBoundingBox.translate(-offsetCenter)
+                                                        )
+                                                    ) {
+                                                        change.consume()
+                                                        appyxComponent.onDrag(dragAmount, density)
+                                                        true
+                                                    } else {
+                                                        appyxComponent.onDragEnd()
+                                                        false
+                                                    }
+                                                },
+                                                onDragEnd = {
+                                                    appyxComponent.onDragEnd()
+                                                },
+                                            )
+                                        }
+                                        .offset { -elementOffset }
+                                        .then(elementUiModel.modifier)
+                                        .onPlaced {
+                                            elementSize = it.size
+                                            val localCenter = Offset(
+                                                it.size.width.toFloat(),
+                                                it.size.height.toFloat()
+                                            ) / 2f
+                                            transformedBoundingBox =
+                                                it
+                                                    .boundsInParent()
+                                                    .inflate(gestureExtraTouchAreaPx)
+                                            offsetCenter = transformedBoundingBox.center - localCenter
+                                        }
+                                ),
+                                saveableStateHolder = saveableStateHolder,
+                                decorator = block
+                            )
+                        }
                     }
                 }
             }
     }
+
+    @Composable
+    private fun elementOffset(
```

Comment:
What about moving this `elementOffset` composable somewhere common but internal?

Reply:
As part of #534 we can keep only one rather than moving out.